OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -190,14 +190,21 @@ Action InitDisplayPanel(Widget panel)
 
 			var frameLimitTextfield = panel.Get<TextFieldWidget>("FRAME_LIMIT_TEXTFIELD");
 			frameLimitTextfield.Text = ds.MaxFramerate.ToString();
-			frameLimitTextfield.OnLoseFocus = () =>
+			frameLimitTextfield.OnEnterKey = () =>
 			{
 				int fps;
 				Exts.TryParseIntegerInvariant(frameLimitTextfield.Text, out fps);
 				ds.MaxFramerate = fps.Clamp(1, 1000);
 				frameLimitTextfield.Text = ds.MaxFramerate.ToString();
+				frameLimitTextfield.YieldKeyboardFocus();
+				return true;
+			};
+			frameLimitTextfield.OnEscKey = () =>
```

Comment:
Ok this also happens in Input settings menu with hotkey assignment. If you keep clicking on different hotkey fields, they all get added to the stack and when you press ESC you switch focus in reverse order. So there has to be a way to prevent other text fields in the same widget from getting focus. Or the previous widget should be removed from stack at TakeKeyboardFocus() if they are in the same parent widget?